From 6176d8ef8f36290075c5bd8932d623e68468a1cd Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Tue, 16 Nov 2021 11:05:17 +0100
Subject: [PATCH] Do updateTheme after QQuickWindow::event

doing updatetheme there will ensure that the window size is correct,
so then when the surface commit will happen, we'll have the blur
and contrast regions wit hthe proper geometry

BUG:305247
---
 src/plasmaquick/dialog.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plasmaquick/dialog.cpp b/src/plasmaquick/dialog.cpp
index eac7d432a..343bfe4c0 100644
--- a/src/plasmaquick/dialog.cpp
+++ b/src/plasmaquick/dialog.cpp
@@ -1243,10 +1243,12 @@ bool Dialog::event(QEvent *event)
         // sometimes non null regions arrive even for non visible windows
         // for which surface creation would fail
         if (!d->shellSurface && isVisible()) {
+            const bool ret = QQuickWindow::event(event);
             KWindowSystem::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
             d->setupWaylandIntegration();
             d->updateVisibility(true);
             d->updateTheme();
+            return ret;
         }
 #endif
     } else if (event->type() == QEvent::PlatformSurface) {
-- 
GitLab

